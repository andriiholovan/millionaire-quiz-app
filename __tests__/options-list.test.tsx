import { expect, vi, test } from 'vitest';
import { render, screen } from '@testing-library/react';
import OptionsList from '../app/_components/options-list';

vi.mock('next/navigation', async () => {
  const actual = await vi.importActual('next/navigation');
  return {
    ...actual,
    useRouter: vi.fn(() => ({
      push: vi.fn(),
    })),
    useParams: vi.fn(() => ({ id: 1 })),
  };
});

vi.mock('react-dom', async () => {
  const actual = await vi.importActual('react-dom');
  return {
    ...actual,
    useFormStatus: vi.fn(() => ({ pending: false })),
  };
});

test('Options buttons', () => {
  const answers = [
    { id: 'A', title: 'option 1', isCorrect: true },
    { id: 'B', title: 'option 2', isCorrect: false },
  ];
  render(<OptionsList answers={answers} />);
  const buttons = screen.getAllByRole('button');
  expect(buttons).toHaveLength(2);
});
