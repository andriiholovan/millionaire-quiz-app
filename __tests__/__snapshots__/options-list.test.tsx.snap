// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`OPTION LIST > Should match snapshot 1`] = `
<DocumentFragment>
  <form
    class="_options_group_2ab0b1"
  >
    <button
      class="_answer_2ab0b1"
      name="id"
      type="submit"
      value="A"
    >
      <span
        class="_answer_label_2ab0b1"
      />
      <span
        class="_answer_title_2ab0b1"
      >
        option 1
      </span>
    </button>
    <button
      class="_answer_2ab0b1"
      name="id"
      type="submit"
      value="B"
    >
      <span
        class="_answer_label_2ab0b1"
      />
      <span
        class="_answer_title_2ab0b1"
      >
        option 2
      </span>
    </button>
  </form>
</DocumentFragment>
`;
